Tests do not pass (fails & timeout)

Accordingly to [CI](https://github.com/godzie44/go-uring/blob/79884120fd55186446a1f4215f31244c51b5cf62/.github/workflows/test.yml), the tests do not run on regular basis (to have an idea whether that a local issue or a global one). Running tests locally leads to failed checks because of unmet conditions and timeouts (see output below). The same behavior is observed when the tests are run with enabled x64 atomic flag  (`go test -v -tags amd64_atomic ./...`). 

**Revision:**
```
~/go-uring$ git rev-parse HEAD
79884120fd55186446a1f4215f31244c51b5cf62
```

**OS:**
```
~/go-uring$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 21.10
Release:        21.10
Codename:       impish
```

**Linux:**
```
~/go-uring$ uname -a
Linux ubuntu.localdomain 5.13.0-22-generic #22-Ubuntu SMP Fri Nov 5 13:21:36 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Output:**
```
~/go-uring$ go test -v ./...
?       github.com/godzie44/go-uring/example/echo-server        [no test files]
?       github.com/godzie44/go-uring/example/echo-server-multi-thread   [no test files]
?       github.com/godzie44/go-uring/example/http-server        [no test files]
=== RUN   TestListenerAccept
--- PASS: TestListenerAccept (0.61s)
PASS
ok      github.com/godzie44/go-uring/net        (cached)
=== RUN   TestNetworkReactor
=== RUN   TestNetworkReactor/TestCancelOperation
    net_test.go:90:
                Error Trace:    net_test.go:90
                Error:          An error is expected but got nil.
                Test:           TestNetworkReactor/TestCancelOperation
                Messages:       operation canceled
=== RUN   TestNetworkReactor/TestExecuteWithDeadline
    net_test.go:71:
                Error Trace:    net_test.go:71
                Error:          Should be true
                Test:           TestNetworkReactor/TestExecuteWithDeadline
--- FAIL: TestNetworkReactor (1.06s)
    --- FAIL: TestNetworkReactor/TestCancelOperation (0.49s)
    --- FAIL: TestNetworkReactor/TestExecuteWithDeadline (0.56s)
=== RUN   TestRequestID
--- PASS: TestRequestID (0.00s)
=== RUN   TestReactor
=== RUN   TestReactor/TestCancelOperation
=== RUN   TestReactor/TestExecuteWithDeadline
=== RUN   TestReactor/TestReactorExecuteReadV
--- PASS: TestReactor (2.11s)
    --- PASS: TestReactor/TestCancelOperation (1.04s)
    --- PASS: TestReactor/TestExecuteWithDeadline (1.04s)
    --- PASS: TestReactor/TestReactorExecuteReadV (0.04s)
=== RUN   TestRegistry
--- PASS: TestRegistry (0.57s)
FAIL
FAIL    github.com/godzie44/go-uring/reactor    3.739s
=== RUN   TestProbe
--- PASS: TestProbe (0.00s)
=== RUN   TestIOWQMaxWorkers
    register_test.go:42: Skipped, IORING_REGISTER_IOWQ_MAX_WORKERS not supported
--- SKIP: TestIOWQMaxWorkers (0.02s)
=== RUN   TestAccept
--- PASS: TestAccept (0.00s)
=== RUN   TestAcceptWithSQPoll
--- PASS: TestAcceptWithSQPoll (0.00s)
=== RUN   TestAcceptCancel
--- PASS: TestAcceptCancel (0.01s)
=== RUN   TestAcceptMany
--- PASS: TestAcceptMany (0.06s)
=== RUN   TestAcceptLink
--- PASS: TestAcceptLink (0.20s)
=== RUN   TestAcceptAddr
panic: test timed out after 10m0s

goroutine 15 [running]:
testing.(*M).startAlarm.func1()
        /usr/lib/go-1.17/src/testing/testing.go:1788 +0x8e
created by time.goFunc
        /usr/lib/go-1.17/src/time/sleep.go:180 +0x31

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0000856c0, {0x5d2bde, 0x46b013}, 0x5e3ab0)
        /usr/lib/go-1.17/src/testing/testing.go:1307 +0x375
testing.runTests.func1(0xc000095290)
        /usr/lib/go-1.17/src/testing/testing.go:1598 +0x6e
testing.tRunner(0xc0000856c0, 0xc00010bd18)
        /usr/lib/go-1.17/src/testing/testing.go:1259 +0x102
testing.runTests(0xc000102080, {0x741ec0, 0x18, 0x18}, {0x47c30d, 0x5d2e00, 0x747140})
        /usr/lib/go-1.17/src/testing/testing.go:1596 +0x43f
testing.(*M).Run(0xc000102080)
        /usr/lib/go-1.17/src/testing/testing.go:1504 +0x51d
main.main()
        _testmain.go:89 +0x14b

goroutine 13 [syscall]:
syscall.Syscall6(0x1aa, 0xf, 0x0, 0x1, 0x1, 0x0, 0x8)
        /usr/lib/go-1.17/src/syscall/asm_linux_amd64.s:43 +0x5
github.com/godzie44/go-uring/uring.sysEnter2(0xc00019c220, 0x612280, 0x0, 0x1a0340, 0x0, 0x0, 0x0)
        /home/vagrant/go-uring/uring/syscall.go:68 +0xa5
github.com/godzie44/go-uring/uring.(*Ring).getCQEvents(0xc00019c220, {0x0, 0x1, 0x0, 0x0, 0x8})
        /home/vagrant/go-uring/uring/ring.go:316 +0x10b
github.com/godzie44/go-uring/uring.(*Ring).WaitCQEvents(...)
        /home/vagrant/go-uring/uring/ring.go:380
github.com/godzie44/go-uring/uring.TestAcceptAddr(0xc0001a0340)
        /home/vagrant/go-uring/uring/ring_accept_test.go:358 +0x2d2
testing.tRunner(0xc0001a0340, 0x5e3ab0)
        /usr/lib/go-1.17/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
        /usr/lib/go-1.17/src/testing/testing.go:1306 +0x35a

goroutine 14 [chan send]:
github.com/godzie44/go-uring/uring.dial(0xc0001a0340, 0x5e3ab0)
        /home/vagrant/go-uring/uring/ring_accept_test.go:23 +0x95
created by github.com/godzie44/go-uring/uring.TestAcceptAddr
        /home/vagrant/go-uring/uring/ring_accept_test.go:349 +0x1b4
FAIL    github.com/godzie44/go-uring/uring      600.148s
FAIL
```

--- a/net/listener_test.go
+++ b/net/listener_test.go
@@ -13,7 +13,7 @@ import (
 )
 
 func TestListenerAccept(t *testing.T) {
-	r, err := uring.New(uring.MaxEntries >> 1)
+	r, err := uring.New(uring.MaxEntries >> 2)
 	require.NoError(t, err)
 	defer r.Close()
 

--- a/reactor/net_test.go
+++ b/reactor/net_test.go
@@ -53,7 +53,7 @@ func (ts *NetworkReactorTestSuite) TearDownTest() {
 }
 
 func (ts *NetworkReactorTestSuite) TestExecuteWithDeadline() {
-	l, fd, err := makeTCPListener("0.0.0.0:8080")
+	l, fd, err := makeTCPListener("0.0.0.0:8081")
 	ts.Require().NoError(err)
 	defer l.Close()
 
@@ -72,7 +72,7 @@ func (ts *NetworkReactorTestSuite) TestExecuteWithDeadline() {
 }
 
 func (ts *NetworkReactorTestSuite) TestCancelOperation() {
-	l, fd, err := makeTCPListener("0.0.0.0:8080")
+	l, fd, err := makeTCPListener("0.0.0.0:8082")
 	ts.Require().NoError(err)
 	defer l.Close()
 

--- a/reactor/reactor_test.go
+++ b/reactor/reactor_test.go
@@ -74,7 +74,7 @@ func (ts *ReactorTestSuite) TestReactorExecuteReadV() {
 }
 
 func (ts *ReactorTestSuite) TestExecuteWithDeadline() {
-	l, fd, err := makeTCPListener("0.0.0.0:8080")
+	l, fd, err := makeTCPListener("0.0.0.0:8083")
 	ts.Require().NoError(err)
 	defer l.Close()
 
@@ -94,7 +94,7 @@ func (ts *ReactorTestSuite) TestExecuteWithDeadline() {
 }
 
 func (ts *ReactorTestSuite) TestCancelOperation() {
-	l, fd, err := makeTCPListener("0.0.0.0:8080")
+	l, fd, err := makeTCPListener("0.0.0.0:8084")
 	ts.Require().NoError(err)
 	defer l.Close()
 

--- a/uring/ring_accept_test.go
+++ b/uring/ring_accept_test.go
@@ -16,8 +16,8 @@ import (
 	"time"
 )
 
-func dial(t *testing.T, connChan chan<- net.Conn) {
-	c, err := net.Dial("tcp", "0.0.0.0:8080")
+func dial(t *testing.T, addr string, connChan chan<- net.Conn) {
+	c, err := net.Dial("tcp", addr)
 	require.NoError(t, err)
 
 	connChan <- c
@@ -61,7 +61,7 @@ func TestAccept(t *testing.T) {
 	require.NoError(t, err)
 	defer ring.Close()
 
-	testAccept(t, ring)
+	testAccept(t, ring, "0.0.0.0:8081")
 }
 
 func TestAcceptWithSQPoll(t *testing.T) {
@@ -72,16 +72,16 @@ func TestAcceptWithSQPoll(t *testing.T) {
 	require.NoError(t, err)
 	defer ring.Close()
 
-	testAccept(t, ring)
+	testAccept(t, ring, "0.0.0.0:8082")
 }
 
-func testAccept(t *testing.T, ring *Ring) {
-	tcpListener, listenerFd, err := makeTCPListener("0.0.0.0:8080")
+func testAccept(t *testing.T, ring *Ring, addr string) {
+	tcpListener, listenerFd, err := makeTCPListener(addr)
 	require.NoError(t, err)
 	defer tcpListener.Close()
 
 	clientConnChan := make(chan net.Conn)
-	go dial(t, clientConnChan)
+	go dial(t, addr, clientConnChan)
 
 	connFd, err := acceptConnection(t, ring, listenerFd)
 	require.NoError(t, err)
@@ -161,11 +161,13 @@ func TestAcceptCancel(t *testing.T) {
 		{time.Microsecond * 10000},
 	}
 
+	const addr = "0.0.0.0:8083"
+
 	for _, tc := range testCases {
 		ring, err := New(32)
 		require.NoError(t, err)
 
-		tcpListener, listenerFd, err := makeTCPListener("0.0.0.0:8080")
+		tcpListener, listenerFd, err := makeTCPListener(addr)
 		require.NoError(t, err)
 
 		require.NoError(t, ring.QueueSQE(Accept(listenerFd, 0), 0, 1))
@@ -274,14 +276,14 @@ func TestAcceptLink(t *testing.T) {
 		wg.Add(1)
 		go func(tc linkTestCase) {
 			defer wg.Done()
-			recvG(t, 8080, tc, syncChan)
+			recvG(t, 8084, tc, syncChan)
 		}(tc)
 
 		if tc.doConnect {
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
-				sendG(t, 8080, syncChan)
+				sendG(t, 8084, syncChan)
 			}()
 		}
 
@@ -341,12 +343,13 @@ func TestAcceptAddr(t *testing.T) {
 	require.NoError(t, err)
 	defer ring.Close()
 
-	tcpListener, listenerFd, err := makeTCPListener("0.0.0.0:8080")
+	const addr = "0.0.0.0:8085"
+	tcpListener, listenerFd, err := makeTCPListener(addr)
 	require.NoError(t, err)
 	defer tcpListener.Close()
 
 	clientConnChan := make(chan net.Conn)
-	go dial(t, clientConnChan)
+	go dial(t, addr, clientConnChan)
 
 	op := Accept(listenerFd, 0)
 

--- a/uring/ring_send_recv_test.go
+++ b/uring/ring_send_recv_test.go
@@ -36,7 +36,7 @@ func send(t *testing.T) {
 	require.NoError(t, err)
 	defer ring.Close()
 
-	conn, err := net.Dial("udp", "127.0.0.1:8080")
+	conn, err := net.Dial("udp", "127.0.0.1:8087")
 	require.NoError(t, err)
 	defer conn.Close()
 
@@ -54,7 +54,7 @@ func recv(t *testing.T, cond *sync.Cond) {
 	require.NoError(t, err)
 	defer ring.Close()
 
-	pc, err := net.ListenPacket("udp", ":8080")
+	pc, err := net.ListenPacket("udp", ":8087")
 	require.NoError(t, err)
 	defer pc.Close()
 

